src/App.js

```diff
@@ -34,50 +34,130 @@ class App extends Component {
         this.changeValue = this.changeValue.bind(this);
         this.setZipLocation = this.setZipLocation.bind(this);
         this.applyFilter = this.applyFilter.bind(this);
+        this.distFrom = this.distFrom.bind(this);
+        this.toRad = this.toRad.bind(this);
+        this.tuitionFilter = this.tuitionFilter.bind(this);
+        this.satFilter = this.satFilter.bind(this);
+        this.actFilter = this.actFilter.bind(this);
+        this.rankingFilter = this.rankingFilter.bind(this);
+
     }
     componentWillMount() {
         var childState = Object.assign({}, this.state);
-       // childState.data = data;
         this.child = React.cloneElement(this.props.children, childState);
     }
     componentWillReceiveProps(props){
         var childState = Object.assign({}, this.state);
-        //childState.data = data;
         this.child = React.cloneElement(this.props.children, childState);
-        //this.forceUpdate();
     }
     setZip(event){
 
         this.handleChange(event);
     }
     changeValue(e){
         this.state.filter.tuition = e.target.value;
-        console.log(this.state);
     }
     applyFilter(e){
         e.preventDefault();
+        if(/(^\d{5}$)|(^\d{5}-\d{4}$)/.test(this.state.filter.zip)){
+            var url = "https://maps.googleapis.com/maps/api/geocode/json?address="+ encodeURIComponent(this.state.filter.zip);
+            fetch(url) //download the data
+            .then(function(res) { return res.json(); })
+            .then((datas) =>{
+                return this.setZipLocation(datas.results);
+            })
+            .then(this.satFilter)
+            .then(this.actFilter)
+            .then(this.rankingFilter)
+            .then(this.tuitionFilter);
+        }else{
+            this.tuitionFilter(this.rankingFilter(this.actFilter(this.satFilter(data))));
```

Comment:
I think there's a better way to do this if/else. Here's how I'd do it:
- Make a variable for the ZIP promise.
- Then do an if/else. If there is a ZIP code, set the ZIP promise. If not, set the ZIP promise to `Promise.resolve(data)`.
- Then outside the if/else, you should be able to use your variable with the string of `.then`s you have.

This way you won't have to repeat the set of operations you want to do twice.